sdk/test-utils/recorder/src/utils/index.ts

```diff
@@ -436,8 +458,74 @@ export function handleSingleQuotesInUrlPath(fixture: string): string {
   return updatedFixture;
 }
 
+/**
+ * Meant for node recordings only!
+ *
+ * Masks access tokens in the json response from nock fixtures.
+ * For example, the following part of the nock fixture/recording would be updated.
+ * from `.reply(200, {"token_type":"Bearer","expires_in":86399,"access_token":"e6z-9_g"}, [`
+ * to   `.reply(200, {"token_type":"Bearer","expires_in":86399,"access_token":"access_token"}, [`
+ *
+ * @param {string} fixture
+ */
+export function maskAccessTokenInNockFixture(fixture: string): string {
+  if (isBrowser()) {
+    throw new Error(
+      `"maskAccessTokenInNockFixture" method is not meant to be used in the browsers`
+    );
+  }
+  // Replaces only if the content-type is json
+  if (isContentTypeInNockFixture(fixture, jsonContentTypes)) {
+    // Matches the nock's reply from the fixture such as below
+    //   `.reply(200, {"token_type":"Bearer","expires_in":86399,"access_token":"e6z-9_g"}, [`
+    const matches = fixture.match(/\.reply\((.*), (.*), .*/);
+    if (matches && matches[2]) {
+      return fixture.replace(/"access_token"\s*:\s*"(.+?)"/, `"access_token":"access_token"`);
+    }
+  }
+  return fixture;
+}
+
+/**
+ * Meant for browser recordings only!
+ *
+ * Masks access tokens in the json recordings of the browser.
+ * For example, the following part of the nock fixture/recording would be updated.
+ * from `"response": "{"token_type":"Bearer","expires_in":86399,"ext_expires_in":86399,"access_token":"e6z-9_g"}",`
+ * to   `"response": "{"token_type":"Bearer","expires_in":86399,"ext_expires_in":86399,"access_token":"access_token"}",`
+ *
+ */
+export function maskAccessTokenInBrowserRecording(fixtures: string): string {
+  if (!isBrowser()) {
+    throw new Error(
+      `"maskAccessTokenInBrowserRecording" method is meant to be used in the browsers only`
+    );
+  }
+
+  // fixture is supposed to be an array of JSON recordings at this point
+  for (let i = 0; i < fixtures.length; i++) {
+    // Replaces only if the content-type is json
+    if (isContentTypeInBrowserRecording(fixtures[i], jsonContentTypes)) {
+      if ((fixtures[i] as any).response) {
+        const parsedResponse = JSON.parse((fixtures[i] as any).response);
+        if (parsedResponse["access_token"]) {
+          parsedResponse["access_token"] = "access_token";
+          (fixtures[i] as any).response = JSON.stringify(parsedResponse);
+        }
+      }
+    }
+  }
+  return fixtures;
+}
+
 /**
  * List of binary content types.
  * Currently, "avro/binary" is the only one present.
  */
 export const binaryContentTypes = ["avro/binary"];
+
+/**
+ * List of json content types.
+ * // TODO: Add anything else to the list??
+ */
+export const jsonContentTypes = ["application/json;charset=utf-8"];
```

Comment:
The recording has a space after `;`. does it matter?

I've also seen "application/json", without the charset part.

Reply:
metrics advisor is sending  PATCH requests with `application/merge-patch+json`

Reply:
> The recording has a space after ;. does it matter?

I'm removing the spaces while matching, so this should be good.

Reply:
> metrics advisor is sending PATCH requests with application/merge-patch+json

Sounds good, can be added.

Reply:
Can you share an example, @jeremymeng?
I couldn't find that in any of the recordings.

Reply:
Now that you asked, I took a look at the recordings and found that they are also sent out using application/json.